includes/admin/home/notices/class-sensei-home-notices.php

```diff
@@ -0,0 +1,317 @@
+	 *
+	 * @param string $plugin_slug    The plugin slug.
+	 * @param string $latest_version The latest known version of the plugin.
+	 *
+	 * @return bool
+	 */
+	protected function is_plugin_update_available( $plugin_slug, $latest_version ) {
+		$available_updates = $this->get_local_plugin_updates();
+
+		return isset( $available_updates[ $plugin_slug ] ) && $latest_version === $available_updates[ $plugin_slug ];
+	}
+
+	/**
+	 * Get the plugin updates that are locally known. Unlicensed plugins should never be known.
+	 *
+	 * @return array
+	 */
+	private function get_local_plugin_updates() {
+		if ( ! isset( $this->local_plugin_updates ) ) {
+			$this->local_plugin_updates = [];
+			require_once ABSPATH . 'wp-admin/includes/update.php';
+
+			$available_updates = get_plugin_updates();
+			foreach ( $available_updates as $plugin_data ) {
+				$plugin_slug    = $plugin_data->update->slug;
+				$update_version = $plugin_data->update->new_version;
+				$update_package = $plugin_data->update->package;
+
+				if ( ! $plugin_slug || ! $update_version || ! $update_package ) {
+					continue;
+				}
+
+				$this->local_plugin_updates[ $plugin_slug ] = $update_version;
+			}
+		}
+
+		return $this->local_plugin_updates;
+	}
+
+	/**
+	 * Get the base settings for a plugin update notice.
+	 *
+	 * @param array $plugin_data The plugin update data.
+	 *
+	 * @return array
+	 */
+	private function get_base_plugin_notice( $plugin_data ) {
+		$changelog_url = $plugin_data['changelog'] ?? false;
+
+		$info_link = false;
+		if ( ! empty( $changelog_url ) ) {
+			$info_link = [
+				'label' => esc_html__( 'What\'s new', 'sensei-lms' ),
+				'url'   => esc_url( $changelog_url ),
+			];
+		}
+
+		// We only want this to be dismissible if Sensei LMS is active and available because it can hande the dismiss requests.
+		$is_dismissible = class_exists( 'Sensei_Admin_Notices' );
+
+		return [
+			'level'       => 'info',
+			'type'        => 'site-wide',
+			'info_link'   => $info_link,
+			'conditions'  => [
+				[
+					'type'    => 'screens',
+					'screens' => [ Sensei_Home::SCREEN_ID ],
+				],
+			],
+			'dismissible' => $is_dismissible,
+			'actions'     => [],
+		];
+	}
+
+	/**
+	 * Get the settings for a plugin update notice on an unlicensed plugin.
+	 *
+	 * @param array $plugin_data The plugin update data.
+	 *
+	 * @return array
+	 */
+	private function get_unlicensed_plugin_update_notice( array $plugin_data ):array {
+		$plugin_name    = $plugin_data['name'];
+		$latest_version = $plugin_data['latest_version'];
+
+		$notice            = $this->get_base_plugin_notice( $plugin_data );
+		$notice['message'] = wp_kses(
+			sprintf(
+				// translators: First placeholder is the plugin name; second placeholder is the latest version.
+				__( 'There is a new version of <strong>%1$s</strong> available (%2$s). Please activate the plugin license in order to proceed with the update process.', 'sensei-lms' ),
```

Comment:
We don't provide any action. The end result we expect is to show at the same time the "License activation" form in the header?

Reply:
Yeah, I thought since the action is [below?] the notice, the action would be presented to them. However, we might eventually want to put these license management screens somewhere else so we can add a link then.